app/settings/tokens/-components/edit-form/component.ts

```diff
@@ -66,12 +66,15 @@ export default class TokenForm extends Component {
 
     @action
     async deleteToken() {
-        try {
-            await this.token!.destroyRecord();
-            this.toast.success(this.intl.t('settings.tokens.deleted'));
-            this.router.transitionTo('settings.tokens');
-        } catch {
-            this.toast.error(this.intl.t('settings.tokens.delete-error'));
+        if (this.token) {
+            try {
+                this.token.deleteRecord();
+                await this.token.save();
+                this.toast.success(this.intl.t('settings.tokens.deleted'));
+                this.router.transitionTo('settings.tokens');
+            } catch {
+                this.toast.error(this.intl.t('settings.tokens.delete-error'));
+            }
```

Comment:
Why is this change here? Was there a problem with the delete token flow? I didn't see anything in the PR description about that.

Reply:
Agreed, this issue only came up after rendering the edit form module. It can be made into a separate ticket, which is likely recommended moving forward.

Specifically, the delete button will hang and then timeout when clicking it when editing the scope. The UI will then become unresponsive until reload.